hooks/pre/10-reset_data.rb

```diff
@@ -2,58 +2,49 @@
 require 'English'
 
 def reset
-  if foreman_installed?
-
-    stop_services
-    start_services(['postgresql'])
-    reset_database
-    reset_candlepin
-    reset_pulp
-    reset_pulpcore if pulpcore_enabled?
-
-  else
-    Kafo::KafoConfigure.logger.warn 'Katello not installed yet, can not drop database!'
-  end
-end
-
-def foreman_installed?
-  `which foreman-rake > /dev/null 2>&1`
-  $CHILD_STATUS.success?
-end
-
-def load_foreman_config
+  stop_services
+  start_services(['postgresql']) if local_postgresql?
+  empty_db_in_postgresql('foreman') if foreman_server?
+  reset_candlepin if module_enabled?('katello')
+  reset_pulp if pulp_enabled?
+  empty_db_in_postgresql('pulpcore') if pulpcore_enabled?
+end
+
+def load_db_config(db)
+  param_helper = if db == 'foreman'
```

Comment:
I still question whether it isn't easier to have 3 methods. If you do go with this method, I'd go with `case`:
```ruby
param_helper = case db
               when 'foreman'
                 {} # full thing
               when 'candlepin'
                 {} # full thing
               when 'pulpcore'
                 {} # full thing
               else
                 raise
               end
```

I'd also consider not adding a `param_helper` hash but assign 4 variables.

Reply:
This seems a lot more readable. Thanks